src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/callback/ExceptionRecordValidatorTest.java

```diff
@@ -0,0 +1,236 @@
+        };
+    }
+
+    @ParameterizedTest(name = "{0}: input:{1} error:{2}")
+    @MethodSource("invalidCaseDataTestParams")
+    void should_return_errors_for_invalid_case_data(String caseReason, CaseDetails input, String error) {
+        checkValidation(
+            input,
+            false,
+            VALIDATOR::getValidation,
+            error
+        );
+    }
+
+    private static Object[][] invalidDocumentTestParams() {
+        String documentTypeError = "Invalid scannedDocuments format. Error: No enum constant.*DocumentType.*";
+        String invalidDateError = "Invalid scannedDocuments format. Error: [Tt]ext.*(could not be parsed)?.*";
+        return new Object[][]{
+            {"Invalid document type", caseWithScannedDocumentData("type", "invalid-type"), documentTypeError},
+            {"Invalid document scanned date", caseWithScannedDocumentData("scannedDate", "date"), invalidDateError},
+            {"Invalid document delivery date", caseWithScannedDocumentData("deliveryDate", "date"), invalidDateError},
+            {"Invalid scanned date format", caseWithScannedDocumentData("scannedDate", "010101"), invalidDateError},
+            {"Invalid delivery date format", caseWithScannedDocumentData(
+                "deliveryDate",
+                "01-01-122"
+            ), invalidDateError},
+            {"Null delivery date format", caseWithScannedDocumentData("deliveryDate", null), invalidDateError},
+            {"Null scanned date format", caseWithScannedDocumentData("scannedDate", null), invalidDateError},
+        };
+    }
+
+    @ParameterizedTest(name = "{0}: input:{1} error:{2}")
+    @MethodSource("invalidDocumentTestParams")
+    void should_return_errors_for_invalid_case_documents(String caseReason, CaseDetails input, String error) {
+        checkValidationErrorMatches(
+            input,
+            VALIDATOR::getValidation,
+            error
+        );
+    }
+
+    @Test
+    void should_throw_exception_when_envelopeId_is_missing() {
+        CaseDetails caseDetails = caseWithData("envelopeId", null);
+
+        assertThatCode(() -> VALIDATOR.getValidation(caseDetails))
+            .isInstanceOf(UnprocessableCaseDataException.class)
+            .hasMessage("Exception record is lacking envelopeId field");
+    }
+
+    private void checkValidation(
+        CaseDetails input,
+        boolean valid,
+        Function<CaseDetails, Validation<Seq<String>, ExceptionRecord>> validationMethod,
+        String errorString
+    ) {
+        Validation<Seq<String>, ExceptionRecord> validation = validationMethod.apply(input);
+
+        if (valid) {
+            assertExceptionRecordMappings(validation);
+        } else {
+            assertSoftly(softly -> {
+                softly.assertThat(validation.isValid()).isFalse();
+                List<String> errors = validation.getError().toList();
+                softly.assertThat(errors).contains(errorString);
+            });
+        }
+    }
```

Comment:
This method is called only 2 times.
Once with `valid = true` and once with `valid = false`.
You can just put the respective if-else blocks in the tests and it will be easier to see what's actually being tested.

Reply:
Moved the code to the tests and removed the method